libminifi/include/utils/MinifiConcurrentQueue.h

```diff
@@ -81,25 +82,38 @@ class ConcurrentQueue {
     : queue_( std::move(other.queue_) ) {}
 
  protected:
-  bool tryDequeue(std::unique_lock<std::mutex>& lck, T& out) {
+  void checkLock(std::unique_lock<std::mutex>& lck) const {
     if (!lck.owns_lock()) {
-      throw std::logic_error("Caller of protected ConcurrentQueue::tryDequeue should own the lock!");
+      throw std::logic_error("Caller of protected functions of ConcurrentQueue should own the lock!"); 
     }
+  }
+
+  bool tryDequeueImpl(std::unique_lock<std::mutex>& lck, T& out) {
+    checkLock(lck);
     if (queue_.empty()) {
       return false;
     }
     out = std::move(queue_.front());
     queue_.pop_front();
     return true;
   }
```

Comment:
I like that you used protected functions taking `std::unique_lock` to make it an error to use the non-locking functions without a lock.